### ISSUE TITLE
Clang warning in FWRecoGeometryESProducer.cc

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#27100]
In CMSSW_11_0_X_2019-04-2300 a new clang warning is reported in FWRecoGeometryESProducer.cc
```
>> Compiling  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/e7c946a97459b4581596e57a49f92b69/opt/cmssw/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_CLANG_X_2019-06-04-2300/src/Fireworks/Geometry/src/FWTGeoRecoGeometryRecord.cc 
/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/llvm/7.1.0/bin/clang++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=70401 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DCMSSW_GIT_HASH='CMSSW_11_0_CLANG_X_2019-06-04-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_11_0_CLANG_X_2019-06-04-2300' -I/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/e7c946a97459b4581596e57a49f92b69/opt/cmssw/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_CLANG_X_2019-06-04-2300/src -I/build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/e7c946a97459b4581596e57a49f92b69/opt/cmssw/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_CLANG_X_2019-06-04-2300/poison -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/cms/cmssw/CMSSW_11_0_CLANG_X_2019-06-02-0000/src -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/lcg/root/6.14.09-pffndh/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/libpng/1.6.16-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/pcre/8.37-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/boost/1.67.0-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/bz2lib/1.0.6-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/clhep/2.4.0.0-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/gsl/2.2.1-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/libjpeg-turbo/2.0.1-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/libuuid/2.22.2-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/sigcpp/2.6.2-pafccj/include/sigc++-2.0 -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/tbb/2019_U3-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/cms/vdt/0.4.0-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/xerces-c/3.1.3-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/xz/5.2.2-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/zlib-x86_64/1.2.11-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/md5/1.0.0-pafccj/include -I/build/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc700/external/tinyxml2/6.2.0-pafccj/include -O2 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++1z -ftree-vectorize -Wstrict-overflow -Werror=array-bounds -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -msse3 -felide-constructors -fmessage-length=0 -Wall -Wno-long-long -Wreturn-type -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-c99-extensions -Wno-c++11-narrowing -D__STRICT_ANSI__ -Wno-unused-private-field -Wno-unknown-pragmas -Wno-unused-command-line-argument -Wno-unknown-warning-option -ftemplate-depth=512 -Wno-error=potentially-evaluated-expression -Wno-error=unused-variable -Wno-error=unused-variable -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS   -fPIC      -MMD -MF tmp/slc7_amd64_gcc700/src/Fireworks/Geometry/src/FireworksGeometry/FWTGeoRecoGeometryRecord.cc.d /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/e7c946a97459b4581596e57a49f92b69/opt/cmssw/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_CLANG_X_2019-06-04-2300/src/Fireworks/Geometry/src/FWTGeoRecoGeometryRecord.cc -o tmp/slc7_amd64_gcc700/src/Fireworks/Geometry/src/FireworksGeometry/FWTGeoRecoGeometryRecord.cc.o
  /build/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/e7c946a97459b4581596e57a49f92b69/opt/cmssw/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_11_0_CLANG_X_2019-06-04-2300/src/Fireworks/Geometry/src/FWRecoGeometryESProducer.cc:317:16: warning: unused variable 'id' [-Wunused-variable]
       GEMDetId id(1, 1, 2, 1, 1, 1);
               ^
1 warning generated.
```

Looking at the code the declaration at line https://github.com/cms-sw/cmssw/blob/master/Fireworks/Geometry/src/FWRecoGeometryESProducer.cc#L317 seems to be effectively unused. To be clarified by experts what is the desired code behaviour .

**Comment by fabiocos** [comment]
assign visualization

**Comment by alja** [comment]
@fabiocos Thanks, I will fix the bug.

**Comment by alja** [comment]
@ianna 
Do you know how to test if "GEM endcap station 2" is existing ?

**Comment by fabiocos** [comment]
@alja are you using the ```id``` variable, called nowhere else, just to throw an exception in https://github.com/cms-sw/cmssw/blob/master/DataFormats/MuonDetId/interface/GEMDetId.h#L95 in case this does not exist? From the C++ point of view why not adding a test on the ```station``` method of ```id``` checking whether it is 2 or not?

**Comment by alja** [comment]
@fabiocos @ianna 
To my understanding line https://github.com/cms-sw/cmssw/blob/master/Fireworks/Geometry/src/FWRecoGeometryESProducer.cc#L319 should be:
 m_geomRecord->slaveGeometry(id);

But, to my surprise accessing GEMDetId id(1, 1, 2, 1, 1, 1) throws an exception. 
All I want to do is make a check GEM endcap station 2 exists in the running geometry scenario.

**Comment by fabiocos** [comment]
@alja @jshlee I see that this block fo code has been added by Jason, he might give some suggestion

**Comment by fabiocos** [comment]
The problem has been finally solved by #27310
